In [1]:
import requests
import pandas as pd
import json
import sqlite3
import zipfile

Feature 1: Read 2 types of data into the analysis. API request pulling JSON data = 1st data pull. 

In [2]:
df_zip = zipfile. ZipFile('lou_metro_library_collection.zip')
df = pd.read_csv(df_zip.open('lou_metro_library_collection.csv'))

creating a variable to store the data

In [3]:
art_data = r.json()

In [4]:
df = pd.DataFrame(art_data)

In [5]:
df.head()

departments
0  {'departmentId': 1, 'displayName': 'American D...
1  {'departmentId': 3, 'displayName': 'Ancient Ne...
2  {'departmentId': 4, 'displayName': 'Arms and A...
3  {'departmentId': 5, 'displayName': 'Arts of Af...
4    {'departmentId': 6, 'displayName': 'Asian Art'}

In [6]:
df.describe()

departments
count                                                  19
unique                                                 19
top     {'departmentId': 1, 'displayName': 'American D...
freq                                                    1

We have 19 unique departments to pull data from. What are they?

In [7]:
print (df)

                                          departments
0   {'departmentId': 1, 'displayName': 'American D...
1   {'departmentId': 3, 'displayName': 'Ancient Ne...
2   {'departmentId': 4, 'displayName': 'Arms and A...
3   {'departmentId': 5, 'displayName': 'Arts of Af...
4     {'departmentId': 6, 'displayName': 'Asian Art'}
5   {'departmentId': 7, 'displayName': 'The Cloist...
6   {'departmentId': 8, 'displayName': 'The Costum...
7   {'departmentId': 9, 'displayName': 'Drawings a...
8   {'departmentId': 10, 'displayName': 'Egyptian ...
9   {'departmentId': 11, 'displayName': 'European ...
10  {'departmentId': 12, 'displayName': 'European ...
11  {'departmentId': 13, 'displayName': 'Greek and...
12  {'departmentId': 14, 'displayName': 'Islamic A...
13  {'departmentId': 15, 'displayName': 'The Rober...
14  {'departmentId': 16, 'displayName': 'The Libra...
15  {'departmentId': 17, 'displayName': 'Medieval ...
16  {'departmentId': 18, 'displayName': 'Musical I...
17  {'departmentId': 19, 'di

Okay. I think I want to work with a specific collection next. Medieval Art sounds interesting. Department ID #17. Let's pull that one next. 
Feature 1, Part 2: 2nd API pull for JSON data. 

In [8]:
r2 = requests.get('https://collectionapi.metmuseum.org/public/collection/v1/objects?departmentIds=17')

In [9]:
art_data2=r2.json()

Create another variable to store this new df. 

In [10]:
df2 = pd.DataFrame(art_data2)

Let's see what we have to work with. 

In [11]:
df2.head()

total  objectIDs
0   7142     157612
1   7142     329835
2   7142     329836
3   7142     329837
4   7142     329841

There are 7142 items in this collection. 

In [12]:
df2.tail()

total  objectIDs
7137   7142     477779
7138   7142     707114
7139   7142     818468
7140   7142     818469
7141   7142     818574

Let's pull info on a random object in this collection. 

In [13]:
r3=requests.get('https://collectionapi.metmuseum.org/public/collection/v1/objects/812961')


In [14]:
object_data=r3.json()

What info is available for this random object?

In [15]:
print(object_data)

{'objectID': 812961, 'isHighlight': False, 'accessionNumber': '2018.913.5', 'accessionYear': '2018', 'isPublicDomain': True, 'primaryImage': 'https://images.metmuseum.org/CRDImages/md/original/LC_TR_311_6_2018-s01.jpg', 'primaryImageSmall': 'https://images.metmuseum.org/CRDImages/md/web-large/LC_TR_311_6_2018-s01.jpg', 'additionalImages': [], 'constituents': None, 'department': 'Medieval Art', 'objectName': 'Textile', 'title': 'Wool Roundel with Mythological Animal', 'culture': 'Egyptian', 'period': '', 'dynasty': '', 'reign': '', 'portfolio': '', 'artistRole': '', 'artistPrefix': '', 'artistDisplayName': '', 'artistDisplayBio': '', 'artistSuffix': '', 'artistAlphaSort': '', 'artistNationality': '', 'artistBeginDate': '', 'artistEndDate': '', 'artistGender': '', 'artistWikidata_URL': '', 'artistULAN_URL': '', 'objectDate': '7th–9th century', 'objectBeginDate': 600, 'objectEndDate': 900, 'medium': 'Polychrome wool and undyed linen; tapestry weave', 'dimensions': '8 3/8 × 5 3/16 in. (21.

Let's create a SQLite connection/database to work with next.

In [16]:
conn = sqlite3.connect('art_data2')

Added a SQL connection and created a database file "art_data2".

In [17]:
cursor = conn.cursor()

Create empty SQLite table

In [18]:
art_stuff = """CREATE TABLE art_stuff (
    objectID INTEGER PRIMARY KEY,
    isHighlight VARCHAR(255),
    accessionNumber VARCHAR(255),
    accessionYear VARCHAR(255),
    isPublicDomain VARCHAR(255),
    primaryImage VARCHAR(255),
    primaryImageSmall VARCHAR(255),
    additionalImagesVARCHAR(255),
    constituents VARCHAR(255),
    department VARCHAR(255),
    objectName VARCHAR(255),
    title VARCHAR(255),
    culture VARCHAR(255),
    period VARCHAR(255),
    dynasty VARCHAR(255),
    reign VARCHAR(255),
    portfolio VARCHAR(255),
    artistRole VARCHAR(255),
    artistPrefix VARCHAR(255),
    artistDisplayName VARCHAR(255),
    artistDisplayBio VARCHAR(255),
    artistSuffix VARCHAR(255),
    artistAlphaSort VARCHAR(255),
    artistNationality VARCHAR(255),
    artistBeginDate VARCHAR(255),
    artistEndDate VARCHAR(255),
    artistGender VARCHAR(255),
    artistWikidata_URL VARCHAR(255),
    artistULAN_URL VARCHAR(255),
    objectDate VARCHAR(255),
    objectBeginDate VARCHAR(255),
    objectEndDate VARCHAR(255),
    medium VARCHAR(255),
    dimensions VARCHAR(255),
    measurements VARCHAR(255),
    creditLine VARCHAR(255),
    geographyType VARCHAR(255),
    city VARCHAR(255),
    state VARCHAR(255),
    county VARCHAR(255),
    country VARCHAR(255),
    region VARCHAR(255),
    subregion VARCHAR(255),
    locale VARCHAR(255),
    locus VARCHAR(255),
    excavation VARCHAR(255),
    river VARCHAR(255),
    classification VARCHAR(255),
    rightsAndReproduction VARCHAR(255),
    linkResource VARCHAR(255),
    metadataDate VARCHAR(255),
    repository VARCHAR(255),
    objectURL VARCHAR(255),
    tags VARCHAR(255),
    objectWikidata_URL VARCHAR(255),
    isTimelineWork VARCHAR(255),
    GalleryNumber VARCHAR(255);"""



Add all the data from the Medieval Art dataset to the empty sql database we just created.

In [19]:
df2.to_sql(art_stuff, conn, if_exists='append', index=False)


7142

It has all 7142 items in the new sql database. I can close this SQL connection for now. 

In [20]:
cursor.close()

Adding another dataset in to this analysis next. Pulling "The Cloisters" next.

In [21]:
cloister = requests.get('https://collectionapi.metmuseum.org/public/collection/v1/objects?departmentIds=7')
cloister_data = cloister.json()
cloister_df = pd.DataFrame(cloister_data)
cloister_df.head()

total  objectIDs
0   2340     463192
1   2340     463377
2   2340     463494
3   2340     463551
4   2340     463568

We have another 2340 items in this set to work with. Let's pull another random one to see some data. 

In [22]:
cloister_item=requests.get('https://collectionapi.metmuseum.org/public/collection/v1/objects/470654')

In [23]:
object_data2=cloister_item.json()

In [24]:
print(object_data2)

{'objectID': 470654, 'isHighlight': False, 'accessionNumber': '25.120.1058', 'accessionYear': '1925', 'isPublicDomain': True, 'primaryImage': 'https://images.metmuseum.org/CRDImages/cl/original/sf25-120-1058ab.jpg', 'primaryImageSmall': 'https://images.metmuseum.org/CRDImages/cl/web-large/sf25-120-1058ab.jpg', 'additionalImages': [], 'constituents': None, 'department': 'The Cloisters', 'objectName': 'Support bar', 'title': 'Support Bar', 'culture': 'French', 'period': '', 'dynasty': '', 'reign': '', 'portfolio': '', 'artistRole': '', 'artistPrefix': '', 'artistDisplayName': '', 'artistDisplayBio': '', 'artistSuffix': '', 'artistAlphaSort': '', 'artistNationality': '', 'artistBeginDate': '', 'artistEndDate': '', 'artistGender': '', 'artistWikidata_URL': '', 'artistULAN_URL': '', 'objectDate': 'ca. 1600', 'objectBeginDate': 1595, 'objectEndDate': 1605, 'medium': 'Iron', 'dimensions': 'Overall: 44 1/2 in. (113 cm)', 'measurements': [{'elementName': 'Overall', 'elementDescription': None, '

So I have no idea what this thing is. "Support Bar" is what it's called. Neat. It's French. Also neat. 

In [25]:
conn = sqlite3.connect('art_data2')
cursor = conn.cursor()

In [26]:
cloister_df.to_sql(art_stuff, conn, if_exists='append', index=False)

2340